cosmo_tester/test_suites/test_manager_upgrade/rollback_idempotency_test.py

```diff
@@ -0,0 +1,113 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from contextlib import contextmanager
+from cStringIO import StringIO
+import json
+import sh
+from path import path
+
+
+from manager_upgrade_base import BaseManagerUpgradeTest
+
+
+class ManagerRollbackIdempotencyTest(BaseManagerUpgradeTest):
+    """Rollback is idempotent - nothing will break if we run it several times.
+
+    In case of rollback breaking (randomly, due to eg. network outages),
+    we should be able to simply re-run it to finish the rollback procedure.
+    """
+
+    @contextmanager
+    def break_rollback(self):
+        """Make sure the sanity test on the manager fails during rollback.
+
+        To do this, we'd like to change the sanity app url to a nonexistent
+        one. Note however that rollback will use the url that is stored
+        on the manager in the rollback properties file, so we need to
+        alter that file.
+        """
+        base_dir = path('/opt/cloudify/sanity')
+        fetched_properties = StringIO()
+        fetched_resources = StringIO()
+
+        properties_path = base_dir / 'node_properties_rollback/properties.json'
+        resources_path = base_dir / 'resources_rollback/__resources.json'
```

Comment:
Currently, when performing a rollback the property is used, but the resources.json is still checked! So we could only change properties, but then it would fail with a weird error. Just changing the value in resources does nothing - only the key is checked against, the value isn't used at all.